### PR TITLE
 Allow ability to select what user roles are excluded (1891).

### PR DESCRIPTION
**Pull request by johnPhillips**
## Summary

Addresses issue #1891

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics/components/common/TrackingExclusionSwitches.js [hunk 1]
```diff
@@ -28,23 +28,50 @@ import { __ } from '@wordpress/i18n';
...
+		message = __( 'All logged-in users will be included in Analytics tracking.', 'google-site-kit' );
+	}
+
+	const updateTrackingDisabled = useCallback( ( users, exclude ) => {
+		let trackingDisabledArray;
+		if ( exclude ) {
+			trackingDisabledArray = trackingDisabled.concat( users );
+		} else {
+			trackingDisabledArray = trackingDisabled.filter( ( item ) => item !== users );
+		}
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/common/TrackingExclusionSwitches.js:62]:
```suggestion
		const trackingDisabledArray = exclude
			? trackingDisabled.concat( users )
			: trackingDisabled.filter( ( item ) => item !== users );
```

File: assets/js/modules/analytics/components/common/TrackingExclusionSwitches.js [hunk 2]
```diff
@@ -56,18 +83,30 @@ export default function TrackingExclusionSwitches() {
...
+					<Switch
+						label={ trackingExclusionLabels[ TRACKING_LOGGED_IN_USERS ] }
+						checked={ trackingDisabled.includes( TRACKING_LOGGED_IN_USERS ) }
+						onClick={ onChangeTrackLoggedInUsers }
+						hideLabel={ false }
+					/>
+				</Cell>
+
+				{ ! trackingDisabled.includes( TRACKING_LOGGED_IN_USERS ) &&
+				(
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/common/TrackingExclusionSwitches.js:97]:
```suggestion
				{ ! trackingDisabled.includes( TRACKING_LOGGED_IN_USERS ) && (
```

File: assets/js/modules/analytics/components/common/TrackingExclusionSwitches.js [hunk 3]
```diff
@@ -56,18 +83,30 @@ export default function TrackingExclusionSwitches() {
...
-			<div>
-				<Switch
-					label={ trackingExclusionLabels[ TRACKING_LOGGED_IN_USERS ] }
-					checked={ trackingDisabled.includes( TRACKING_LOGGED_IN_USERS ) }
-					onClick={ onChange }
-					hideLabel={ false }
-				/>
-			</div>
+			<Row>
+				<Cell>
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/common/TrackingExclusionSwitches.js:87]: I think Row and Cell aren't really needed here. Let's wrap <Switche /> elements with just regular divs:

```html
<div>
    <Switch ...>
</div>
<div>
    <Switch ...>
</div>
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Nice work, @johnPhillips. Just a few cosmetic changes and it should be good to go. Could you please address it?

File: assets/js/modules/analytics/components/common/TrackingExclusionSwitches.js (hunk 3 above)
Comment (johnPhillips) [reply assets/js/modules/analytics/components/common/TrackingExclusionSwitches.js:87]: There is an AC about having them aligned right and left:
> The toggle for contentCreators should appear right of the existing toggle for loggedinUsers.

...Using `div` doesn't work, I guess as they are block level elements, so they'd need floating etc. to align them. 

| Using `div` | Using `Row` / `Cell` |
| --- | --- |
|  Screenshot 2021-05-04 at 12 00 40 | Screenshot 2021-05-04 at 12 01 51 |

Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/common/TrackingExclusionSwitches.js:85]: Ah, ok... I did miss this point. Then we need to wrap the row element with `<Grid>` and add appropriate size attributes for cell elements to make sure they are displayed next to each other on desktop and tablet screens, and stacked on mobile devices.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]

**Review by eugene-manuilov — APPROVED** [review]
Looks good to me. Thanks, @johnPhillips!

**Pull request merged**